frontend/app/components/api/api-v3/hal-resources/hal-resource.service.ts

```diff
@@ -100,7 +101,27 @@ export class HalResource {
     this.$initialize($source);
   }
 
+  /**
+   * Return the associated state to this HAL resource, if any.
+   */
+  public get state():State<HalResource>|null {
+    return null;
```

Comment:
I like the line of thought of each object having a reference to it's state and it looks nice in the `SchemaResource`. But it is not used in the WorkPackageResource, I believe because of the `WorkPackageCacheService` which is inconsistent IMO.